packages/profile-sync-controller/src/controllers/user-storage/network-syncing/controller-integration.update-network.test.ts

```diff
@@ -0,0 +1,274 @@
+import { ControllerMessenger } from '@metamask/base-controller';
+import type {
+  NetworkState,
+  NetworkControllerActions,
+} from '@metamask/network-controller';
+import {
+  NetworkController,
+  NetworkStatus,
+  RpcEndpointType,
+} from '@metamask/network-controller';
+import nock from 'nock';
+
+import type { UserStorageControllerMessenger } from '..';
+import type { RPCEndpoint } from './__fixtures__/mockNetwork';
+import {
+  createMockCustomRpcEndpoint,
+  createMockInfuraRpcEndpoint,
+  createMockNetworkConfiguration,
+} from './__fixtures__/mockNetwork';
+import { dispatchUpdateNetwork } from './controller-integration';
+import type { NetworkConfiguration } from './types';
+
+const createNetworkControllerState = (
+  rpcs: RPCEndpoint[] = [createMockInfuraRpcEndpoint()],
+): NetworkState => {
+  const mockNetworkConfig = createMockNetworkConfiguration({ chainId: '0x1' });
+  mockNetworkConfig.rpcEndpoints = rpcs;
+
+  const state: NetworkState = {
+    selectedNetworkClientId: 'mainnet',
+    networkConfigurationsByChainId: {
+      '0x1': mockNetworkConfig,
+    },
+    networksMetadata: {},
+  };
+
+  rpcs.forEach((r) => {
+    state.networksMetadata[r.networkClientId] = {
+      EIPS: {
+        '1559': true,
+      },
+      status: NetworkStatus.Available,
+    };
+  });
+
+  return state;
+};
+
+const createNetworkConfigurationWithRpcs = (rpcs: RPCEndpoint[]) => {
+  const config = createMockNetworkConfiguration({ chainId: '0x1' });
+  config.rpcEndpoints = rpcs;
+  return config;
+};
+
+describe('network-syncing/controller-integration - dispatchUpdateNetwork()', () => {
```

Comment:
I really like this sort of test.
Sure, can argue that this isn't a true integration tests, since we still mocking some stuff, but it is really nice to be able to test actually call the NetworkController to validate the update step for this feature.

I might create other integration parts in another PR - the main one to test was the `NetworkController:updateNetwork`.